src/autopas/AutoPasImpl.h

```diff
@@ -181,8 +211,14 @@ std::vector<Particle> AutoPas<Particle>::updateContainer() {
 template <class Particle>
 std::vector<Particle> AutoPas<Particle>::resizeBox(const std::array<double, 3> &boxMin,
                                                    const std::array<double, 3> &boxMax) {
-  _boxMin = boxMin;
-  _boxMax = boxMax;
+  if (_allowedCellSizeFactors->isInterval()) {
+    AutoPasLog(WARN,
+               "The allowed Cell Size Factors are a continuous interval but internally only those values that "
+               "yield unique numbers of cells are used. Resizing does not cause these values to be recalculated so "
+               "the same configurations might now yield different and non-unique numbers of cells!");
+  }
```

Comment:
Difficult question. So first what exactly do I mean:

If you tell AutoPas that it can choose the cell size factor (CSF) in the continuous interval from [0.5, 3] there can be infinitely many options. However, the actual number of cells in the domain has only a limited number of options. E.g. CSF [0.5, 1[ could map to 25, [1, 2.5[ to 26, and [2.5, 3] to 27 cells per dimension.

When the domain is resized, these mappings might change. This is a potential problem for two reasons:

1. When we encounter continuous CSF we calculate which values lead to unique cells per dimension, so for the example above we would use the discrete CSF options of {0.5, 1, 2.5}, each mapping to a different number of cells. After the resizing, these might not map to the same or even unique numbers of cells.

2. We associate tuning evidence with a fully specified configuration. The CSF is part of the configuration. After resizing, what should we do with the CSF options? If they change, can we associate new options with any existing tuning evidence configurations?

But all of this only matters if you look at continuous CSF...

Reply:
Ah okay, thanks for the in-depth explanation.
So, at some point a recalculation of CSF options might be useful?
I wonder how useful continuous CSFs are anyway...

However, we can keep it with the Warning for now.

Reply:
> I wonder how useful continuous CSFs are anyway...

Useful in the sense that we don't exactly know the relevant values in advance, especially in an MPI simulation where the subdomain size can change. But maybe we want to express the actual information somehow in a different way but that goes beyond the scope of this PR...